shared/loadgenerator/loadgenerator.go

```diff
@@ -109,10 +110,8 @@ func (g *GeneratorOptions) RunLoadTest(resolvableDomain bool) (*GeneratorResults
 // SaveJSON saves the results as Json in the artifacts directory
 func (gr *GeneratorResults) SaveJSON(testName string) error {
 	dir := prow.GetLocalArtifactsDir()
```

Comment:
Is it worth making GetLocalArtifactsDir() creating the dir automatically, if it doesn't exist? Or maybe encapsulate this pattern in a "GetOrCreate" function?

Reply:
Seems it's not worth changing this function, [here](https://github.com/knative/test-infra/blob/4ce16d390c55c71290f3b285bdfe37a1c5490304/tools/flaky-test-reporter/result.go#L118) it's only used for retrieving the path string. And several test cases are also using it for such purpose.